wear/src/main/java/com/android/gpstest/wear/MainActivity.kt

```diff
@@ -0,0 +1,431 @@
+            if (grantResults.isNotEmpty() && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+                userDeniedPermission = false
+            } else {
+                userDeniedPermission = true
+            }
+        }
+    }
+
+    @ExperimentalCoroutinesApi
+    @SuppressLint("MissingPermission")
+    @Synchronized
+    private fun gpsStart() {
+        PreferenceUtils.saveTrackingStarted(true, prefs)
+
+        // Observe flows
+        observeLocationFlow()
+        observeGnssStates()
+    }
+
+    @ExperimentalCoroutinesApi
+    private fun observeLocationFlow() {
+        // This should be a Flow and not LiveData to ensure that the Flow is active before the Service is bound
+        if (locationFlow?.isActive == true) {
+            // If we're already observing updates, don't register again
+            return
+        }
+        // Observe locations via Flow as they are generated by the repository
+        locationFlow = repository.getLocations()
+            .flowWithLifecycle(lifecycle, Lifecycle.State.STARTED)
+            .onEach {
+                //lastLocation = it
+                Log.d(TAG, "Activity location: $it")
+            }
+            .launchIn(lifecycleScope)
+    }
+
+    @OptIn(ExperimentalCoroutinesApi::class)
+    private fun observeGnssStates() {
+        // Use ViewModel here to ensure that it's populated for fragments as well -
+        // otherwise ViewModel is lazily initialized and we don't save TTFF if viewed later in Status (e.g., if started in Accuracy or Map)
+        val gnssStateObserver = Observer<FixState> { fixState ->
+            when (fixState) {
+                is FixState.Acquired -> hideProgressBar()
+                is FixState.NotAcquired -> if (PreferenceUtils.isTrackingStarted(prefs)) showProgressBar()
+            }
+        }
+        signalInfoViewModel.fixState.observe(
+            this, gnssStateObserver
+        )
+    }
+
+    private fun showProgressBar() {
+        val p = progressBar
+        if (p != null) {
+            LibUIUtils.showViewWithAnimation(p, LibUIUtils.ANIMATION_DURATION_SHORT_MS)
+        }
+    }
+
+    private fun hideProgressBar() {
+        val p = progressBar
+        if (p != null) {
+            LibUIUtils.hideViewWithAnimation(p, LibUIUtils.ANIMATION_DURATION_SHORT_MS)
+        }
+    }
+
+    @Synchronized
+    private fun gpsStop() {
+        PreferenceUtils.saveTrackingStarted(false, prefs)
+        locationFlow?.cancel()
+
+        // Reset the options menu to trigger updates to action bar menu items
+        invalidateOptionsMenu()
+        progressBar?.visibility = View.GONE
+    }
+
+    companion object {
+        private const val TAG = "MainActivity"
+    }
+}
+
+@OptIn(ExperimentalCoroutinesApi::class)
+@Composable
+fun WearApp(satStatues: List<String>, signalInfoViewModel: SignalInfoViewModel) {
+    val gnssStatuses: List<SatelliteStatus> by signalInfoViewModel.filteredGnssStatuses.observeAsState(emptyList())
```

Comment:
To get `Location` updates to populate the lat/lon/alt/etc., you'll need another observer here, like:

```kotlin
val location: Location by signalInfoViewModel.location.observeAsState(Location("default"))
```

Then every time the `location` in the ViewModel changes you'll get an update here.

Reply:
> To get `Location` updates to populate the lat/lon/alt/etc., you'll need another observer here, like:
> 
> ```kotlin
> val location: Location by signalInfoViewModel.location.observeAsState(Location("default"))
> ```
> 
> Then every time the `location` in the ViewModel changes you'll get an update here.

Ok, thanks!